src/cpp/rtps/writer/ReaderProxy.cpp

```diff
@@ -681,6 +681,21 @@ ReaderProxy::ChangeConstIterator ReaderProxy::find_change(
            : it->getSequenceNumber() == seq_num ? it : end;
 }
 
+bool ReaderProxy::has_been_delivered(
+        const SequenceNumber_t& seq_number,
+        bool& found) const
+{
+    for (auto change : changes_for_reader_)
+    {
+        if (change.getSequenceNumber() == seq_number)
+        {
+            found = true;
+            return change.has_been_delivered();
+        }
+    }
+    return false;
```

Comment:
No need to traverse the collection:

```suggestion
    if (seq_num <= changes_low_mark_)
    {
        // This means the change has been acknowledged, so it should have been delivered
        return true;
    }

    ChangeIterator it = find_change(seq_num, true);
    if (it != changes_for_reader_.end())
    {
        found = true;
        return change.has_been_delivered();
    }

    return false;
```

Reply:
Done in b60cc43ac and e9cc00b3e